### DISCUSSION TITLE
Route road info

### DISCUSSION THREAD

**Discussion by girishyadav20** [valhalla/valhalla#4693] [Q&A] [answered]
Hi, I am using Valhalla engine to generate routes. How to get road information (road_type, surface, number of lane etc.) for each road segment along the route.

**Comment by dnesbitt61** [comment, marked as answer]
A subsequent call to trace_attributes can be made using the shape/geometry of the route result. This will return attributes for each road segment (edge in the routing graph) on the original route.

**Reply by girishyadav20** [reply]
Thank you @dnesbitt61
